sdk/avalon_sdk/work_order/work_order_request_validator.py

```diff
@@ -70,104 +40,183 @@ def validate_parameters(self, params):
         if len(params) == 0:
             return False, "Empty params in the request"
 
-        key_list = []
-        for key in params.keys():
-            if key not in self.__param_key_map.keys():
-                return False, "Invalid parameter {}".format(key)
+        schema = {
+            "type": "object",
+            "uniqueItems": True,
+            "additionalProperties": False,
+            "properties": {
+                "responseTimeoutMSecs": {
+                    "type": "integer",
+                    "error_msg":
+                    "Invalid data format for responseTimeoutMSecs"},
+                "payloadFormat": {
+                    "type": "string",
+                    "enum": ["JSON-RPC", "json-rpc"],
+                    "error_msg":
+                    "Invalid payload format"},
+                "resultUri": {
+                    "type": "string",
+                    "error_msg":
+                    "Invalid data format for resultUri"},
+                "notifyUri": {
+                    "type": "string",
+                    "error_msg":
+                    "Invalid data format for notifyUri"},
+                "workOrderId": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]{16}",
+                    "error_msg":
+                    "Invalid data format for work order id"},
+                "workerId": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]",
+                    "error_msg":
+                    "Invalid data format for Worker id"},
+                "workloadId": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]",
+                    "error_msg":
+                    "Invalid data format for work load id"},
+                "requesterId": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]{64}",
+                    "error_msg":
+                    "Invalid data format for requester id"},
+                "workerEncryptionKey": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]",
+                    "error_msg":
+                    "Empty or Invalid dataformat for workerEncryptionKey"},
+                "dataEncryptionAlgorithm": {
+                    "type": "string",
+                    "enum": ["AES-GCM-256", "aes-gcm-256"],
+                    "error_msg":
+                    "Invalid data format for dataEncryptionAlgorithm"},
+                "encryptedSessionKey": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]",
+                    "error_msg":
+                    "Invalid data format for encrypted session key"},
+                "sessionKeyIv": {
+                    "type": "string",
+                    "pattern": "[0-9xXa-fA-F]",
+                    "error_msg":
```

Comment:
Add size check wherever it necessary.

Reply:
there is no size mentioned in Spec for direct mode. Hence did not add.